### ISSUE TITLE
Recast and detour applications

### ISSUE THREAD

**Issue by mimooh** [recastnavigation/recastnavigation#333]
A feature request, or perhaps it already exists: It would be lovely to have the two console apps working in such a manner:

./recast -params geometry.obj -o navmesh.nav
./detour -from p0 - to P1 navmesh.nav

**Comment by mimooh** [comment]
Would be even more lovely to have the detour app working as a daemon: loading navmesh once, and wait and answer multiple  "-from p0 -to p1" queries. The results would be the waypoints.

**Comment by jakobbotsch** [comment]
Recast and Detour are mainly libraries that are supposed to be incorporated directly into your application. They are really libraries that support a lot of different kinds of spatial queries -- much more than just finding paths (though this is obviously a popular usage). The demo is not supposed to be something you use directly in your production applications, it is supposed to demonstrate what the libraries can do, and how you can do it.

For example, typically when using Detour you would not compute a full path of points when moving an agent from A to B, because often the path will change underway. Instead you compute as little as possible, and when the path changes you can tell Detour this and it will work it out in a much more efficient way than recomputing a full path.

It would be very hard to shoehorn all the features and good practices of Recast & Detour into easy-to-use executables. Additionally, a main goal of Recast & Detour's is performance, which would be hard to achieve in this manner.

Do you have any specific use-cases for such executables in mind? Recast & Detour is mainly targeted for games, and I am having a hard time of thinking of an example where incurring the overhead of launching new processes for every query would be acceptable.

**Comment by mimooh** [comment]
We are creating simulations of humans evacuating from buildings. Crowds can have up to 200 agents and are locally driven by RVO2. Geometries are 2D, the easy scenario could be:

![a](https://user-images.githubusercontent.com/14321180/41395367-b2518d94-6fad-11e8-8342-2dc42b02f318.png)

Our idea is that initially each agent would get a complete waypoints to the exit. Then the agent would advance from the first waypoint to the next and normally we won't need to recalculate. Sometimes, when things get dense and the agent is pushed out of his track by RVO2 and he cannot see his next waypoint (doesn't happen very often) we would need to recalculate his targets.

We considered various approaches. Navmeshes are the most fancy, but also vector field are good candidates too. For the vector field we would need to have the field generator which assigns the direction for each cell, and recast could be a good tool for such a generator as I imagine.

From my perspective, what would solve our problems would be the two apps which I described. We work in python. For last couple of months we have been considering various approaches: from creating our own python solutions, via trying whatever from github, up to quick c++ courses and basically we had made no progress. So now we are trying the method of asking someone to create such two apps :)
